src/ARVRInterface.cpp

```diff
@@ -330,164 +268,39 @@ void godot_arvr_commit_for_eye(void *p_data, godot_int p_eye, godot_rid *p_rende
-							&transform, true, true);
-
-					// update our button state structure
-					vr::VRControllerState_t old_state = arvr_data->tracked_device_state[i];
-					vr::VRControllerState_t new_state;
-					arvr_data->ovr->hmd->GetControllerState(i, &new_state, sizeof(vr::VRControllerState_t));
-					if (arvr_data->tracked_device_state[i].unPacketNum != new_state.unPacketNum) {
-						// OpenVR supports 5 axis with x/y, for triggers only x is set
-						for (int axis = 0; axis < 5; axis++) {
-							vr::EVRControllerAxisType axis_type =
-									(vr::EVRControllerAxisType)
-											arvr_data->ovr->hmd->GetInt32TrackedDeviceProperty(
-													i, (vr::ETrackedDeviceProperty)(
-															   vr::Prop_Axis0Type_Int32 + axis));
-							if (axis_type == vr::k_eControllerAxis_None) {
-								// not applicable for this controller, ignore
-							} else if (axis_type == vr::k_eControllerAxis_Trigger) {
-								// only x which ranges between 0. and 1.0
-								if (new_state.rAxis[axis].x != old_state.rAxis[axis].x) {
-									godot::arvr_api->godot_arvr_set_controller_axis(
-											arvr_data->trackers[i], axis * 2, new_state.rAxis[axis].x,
-											true); // had some weirdness with this false, I think it may be -1.0 to 1.0 afterall
-								}
-							} else {
-								// this can be trackpad or joystick. Might need to do more with trackpad..
-								if (new_state.rAxis[axis].x != old_state.rAxis[axis].x) {
-									godot::arvr_api->godot_arvr_set_controller_axis(
-											arvr_data->trackers[i], axis * 2, new_state.rAxis[axis].x,
-											true);
-								}
-
-								if (new_state.rAxis[axis].y != old_state.rAxis[axis].y) {
-									godot::arvr_api->godot_arvr_set_controller_axis(
-											arvr_data->trackers[i], (axis * 2) + 1,
-											new_state.rAxis[axis].y, true);
-								}
-							}
-						}
-
-						arvr_data->tracked_device_state[i] = new_state;
-					}
-
-					// update rumble
-					float rumble = godot::arvr_api->godot_arvr_get_controller_rumble(arvr_data->trackers[i]);
-					if ((rumble > 0.0) && ((msec - arvr_data->last_rumble_update[i]) > 5)) {
-						// We should only call this once ever 5ms...
-						arvr_data->ovr->hmd->TriggerHapticPulse(i, 0, (rumble * 5000.0));
-						arvr_data->last_rumble_update[i] = msec;
-					}
-				}
-			}
-		}
 	}
 }
 
+////////////////////////////////////////////////////////////////
+// Construct our interface so it can be registered
+// we do not initialise anything here!
 void *godot_arvr_constructor(godot_object *p_instance) {
 	// note, don't do to much here, not much will have been initialised yet...
 
 	arvr_data_struct *arvr_data = (arvr_data_struct *)godot::api->godot_alloc(sizeof(arvr_data_struct));
 
 	arvr_data->ovr = openvr_data::retain_singleton();
 	arvr_data->video_driver = 0;
-	godot::api->godot_transform_new_identity(&arvr_data->hmd_transform);
+	godot::api->godot_transform_new_identity(&arvr_data->ovr->get_hmd_transform());
```

Comment:
Same error about temporaries here.

Reply:
This line can actually be removed, it's already properly initialized inside of the openvr_data object.